java-checks/src/main/java/org/sonar/java/checks/NullPointerCheck.java

```diff
@@ -453,138 +433,98 @@ void registerAssignedSymbol(Symbol symbol) {
     }
   }
 
-  public enum AbstractValue {
-    UNSET,
-    // value is known to be not null.
-    NOTNULL,
-    // value is known to be null.
-    NULL,
-    // value is unknown (could be null or not null).
-    UNKNOWN
-  }
-
-  @VisibleForTesting
-  static class ConditionalState {
-    final State falseState;
-    final State trueState;
-
-    ConditionalState(State currentState) {
-      falseState = new State(currentState);
-      trueState = new State(currentState);
-    }
-
-    void mergeConditionalAnd(ConditionalState leftConditionalState, ConditionalState rightConditionalState) {
-      // copies the learned values to the parent.
-      trueState.copyValuesFrom(leftConditionalState.trueState);
-      trueState.copyValuesFrom(rightConditionalState.trueState);
-      // invalidates both false states and copies values to the parent
-      falseState.copyValuesFrom(leftConditionalState.falseState.invalidateValues());
-      falseState.copyValuesFrom(rightConditionalState.falseState.invalidateValues());
-    }
-
-    void mergeConditionalOr(ConditionalState leftConditionalState, ConditionalState rightConditionalState) {
-      // invalidates both true states and copies value to the parent.
-      trueState.copyValuesFrom(leftConditionalState.trueState.invalidateValues());
-      trueState.copyValuesFrom(rightConditionalState.trueState.invalidateValues());
-      // copies the learned values to the parent.
-      falseState.copyValuesFrom(leftConditionalState.falseState);
-      falseState.copyValuesFrom(rightConditionalState.falseState);
-    }
-  }
-
-  @VisibleForTesting
-  static class State {
-    @Nullable
-    final State parentState;
-    final Map<VariableSymbol, AbstractValue> variables;
-
-    public State() {
-      this.parentState = null;
-      this.variables = new HashMap<>();
-    }
-
-    public State(State parentState) {
-      this.parentState = parentState;
-      this.variables = new HashMap<>();
+  public class NullPointerVisitor extends SymbolicExecutionCheck {
```

Comment:
This remark applies also to other classes extending SymbolicExecutionCheck (i.e. `LockedVisitor` and `CloseableVisitor`)

I think that the names of these classes are especially confusing. They are definitely NOT visitors compared to what we expect regarding theirs behaviors (at least with respects to `BaseTreeVisitor` and `SubscriptionBaseVisitor`).